sample/samplers.go

```diff
@@ -87,8 +87,8 @@ func (s *Sampler) sample(tokens []token) (token, error) {
 	// topK also sorts the tokens in descending order of logits
 	tokens = topK(tokens, s.topK)
 
-	// token logit values are updated to probabilities
 	tokens = temperature(tokens, s.temperature)
+	tokens = softmax(tokens)
```

Comment:
Any further thoughts on how we could avoid the `func transform(tokens []token) []token` interface that both mutates and returns a slice. This is more of a reminder, it can be in a follow up change. Perhaps we change the interface to

```func transform(tokens *[]token)``` and it can be called as

```softmax(&tokens)``` which visibly edits the slice

Reply:
And FWIW, returning a `[]token` is fine too, but doing so implies that `tokens` may not be mutated. If it ends up being the best scenario to keep it as-is we should add a comment to the ones that mutate arguments so callers know

Reply:
Going to hack on this change in a follow up